crates/noirc_frontend/src/ast/traits.rs

```diff
@@ -68,10 +68,16 @@ pub struct TraitImpl {
     pub items: Vec<TraitImplItem>,
 }
 
-/// Represents a trait constraint such as `where Foo: Display`
+/// Represents a trait constraint such as `where Foo: Display + TraitX + TraitY<U, V>`
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct TraitConstraint {
-    pub typ: UnresolvedType,
+    pub generic_type_name: Ident,
+    pub trait_bounds: Vec<TraitBound>,
+}
```

Comment:
Making a note of this, can we simplify these? E.g. I don't think the compiler would care that the user wrote `T: Foo + Bar` over `T: Foo, T: Bar`, so it would be simpler for the rest of the compiler if we only had the one trait constraint list and didn't need to worry about nested constraints.

Reply:
Indeed, since T can be any type, it makes sense to convert 'T:Foo + Bar' to 'T: Foo, T: Bar'. When I wrote this, I thought T can only be a generic type name, and I thought they couldn't repeat. But apparently, that's not the case. I will fix this as well.

Reply:
Yep, that's what I had in mind in this comment. It's not a required transformation but I think it'd simplify things internally a bit.

Reply:
Ok, done.